lib/tasks/so_version.rake

```diff
@@ -0,0 +1,50 @@
+#--
+# Copyright (C) 2021 SUSE LLC
+#   This library is free software; you can redistribute it and/or modify
+# it only under the terms of version 2.1 of the GNU Lesser General Public
+# License as published by the Free Software Foundation.
+#
+#   This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+# details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#++
+
+namespace :so_version do
+  include Libyui::Tasks::Helpers
+
+  desc "Check that the so version numbers are in sync"
+  task :check do
+    filenames = Dir.glob("package/*.spec").sort
+    so_versions = filenames.each_with_object({}) { |f, h| h[f] = so_version(f) }
+
+    raise "so version mismatch: #{so_versions}" if so_versions.values.compact.uniq.size > 1
+
+    puts so_version if verbose
+  end
+
+  desc "Increase the so version in spec files"
+  task bump: :check do
+    next_so_version = so_version.to_i.next
+
+    Dir.glob("package/*.spec").each do |filename|
+      spec_content = File.read(filename)
+      spec_content.gsub!(
+        /^%define\s+so_version\s+\d+$/,
+        "%define         so_version #{next_so_version}"
```

Comment:
I worry that this can potentially break formatting, why not more careful just digit replacement like
```
      spec_content.gsub!(
        /^(%define\s+so_version\s+)\d+$/,
        "\\1#{next_so_version}"
```